src/ray/stats/metric_defs.cc

```diff
@@ -67,6 +67,21 @@ DEFINE_stats(actors,
              (),
              ray::stats::GAUGE);
 
+/// Job related stats.
+DEFINE_stats(running_jobs,
+             "Number of jobs currently running.",
+             /*tags=*/(),
+             /*buckets=*/(),
+             ray::stats::GAUGE);
+
+DEFINE_stats(finished_jobs,
+             "New finished jobs number.",
+             // TODO(hjiang): Consider adding task completion status, for example, failed,
```

Comment:
Would it be difficult to add the completion status in this PR?

Reply:
This is what I'm thinking now: this PR adds metrics for driver jobs (handled by gcs_job_manager.cc). The next PR we can extend it to support submission job (handled by job_manager.py) as well by introducing a `Type` label. 

So the `jobs` metric has `Status` label (possible values are `class JobStatus(str, Enum):`) (@dentiny this is different from what I told you before with just RUNNING and FINISHED state) (For driver job, the only statuses are `RUNNING` and `SUCCEEDED`), `Type` label (possible values are `class JobType(str, Enum):`).

WDTY @alanwguo @rynewang ?

Reply:
I think there two requirements here, as the discussion goes on:
- (the initial feature requirement) Record the number of running jobs, so alerts could be created on idle cluster
  + I think gauge is the proper metrics type 
- Job state transfer, the typical ones of which are running, completed, etc
  + counter is a proper metrics type

> Would it be difficult to add the completion status in this PR?

For this particular PR, I would like to focus on (1) first, and resolve (2) later, mainly because I could see there're more work to do out of gcs job manager.
For example, I already see a few other [states](https://github.com/ray-project/ray/blob/7591f91832872fc4606dd54641bb2b781de170a0/python/ray/dashboard/modules/job/common.py#L35-L62), thus more metrics uploading in state transfer logic.

> this PR adds metrics for driver jobs (handled by gcs_job_manager.cc). The next PR we can extend it to support submission job (handled by job_manager.py) as well by introducing a Type label.

I basically agree with you: for limited (and important) states, we should have counter for each states increment and decrement.
Within this PR, I add metrics (via counter) for finished jobs; it's useful to measure the system throughput.

Some other counter metrics I'm thinking:
- Successful jobs counter: used to measure system goodput (which is different with throughput)
- Failed jobs counter: used to create alerts, also error status code aggregated should be useful (i.e. leverage grpc status)
- Queued jobs counter: useful as an indicator for system load and input for HPA

Reply:
For this particular PR, I would like to focus on 
- Gauge for running jobs, which is the issue initially asking for
- Counter for completed/finished jobs
  + If you guys prefer to keep it simple, I would remove that part of code

Reply:
@jjyao  this confuses me. do you mean submission jobs are not sent to gcs_job_manager? I thought it needs to keep all jobs that has a job id.

Reply:
Kindly ping, any suggestions for my comments?

Reply:
I think we can first do no-labels; if later we need labels for dead jobs we can do that in another PR

Reply:
Sg, @rynewang let me know if there're other things you want me to change in this PR :)

Reply:
> submission jobs are not sent to gcs_job_manager?

gcs_job_manager doesn't manage submission jobs. A single submission job can have 0-N driver jobs.

Reply:
@dentiny, let me sync up with @rynewang first and get back to you.